### PR TITLE
feat(sidebar): New sidebar title and favorite color area COMPASS-5961

### PR DESCRIPTION
**Pull request by lerouxb**
Screenshot 2022-08-08 at 14 29 35

Screenshot 2022-08-09 at 15 33 33

Screenshot 2022-08-09 at 15 57 56

I stashed most of the dark mode colour changes for now in the nav tree because there are still lots of uncertainties there and it affects the old sidebar as well (ie. without a feature flag).

"Copy connection string" is implemented. "Edit favorite" is implemented. It changes to "Save favorite" if the connection isn't a favorite yet. "Cluster info" is still unimplemented.

### CODE REVIEW

File: packages/compass-components/src/components/item-action-controls.tsx [hunk 1]
```diff
@@ -31,21 +26,21 @@ const actionIconButton = css({
...
 
-export const ActionControls: React.FunctionComponent<{
-  actions: ActionControlOptions;
-  onAction(actionName: Actions): void;
-  className?: string;
-  isActive: boolean;
-  isHovered: boolean;
-  shouldCollapseActionsToMenu?: boolean;
-}> = ({
+export function ItemActionControls<Actions extends string>({
```
Comment (lerouxb) [inline packages/compass-components/src/components/item-action-controls.tsx:29]: @gribnoysup I couldn't figure out how to add the Actions type to make a generic function with the previously used syntax. Is this a problem? Any points on how to do this better?

File: packages/compass-databases-navigation/src/collection-item.tsx [hunk 2]
```diff
@@ -5,18 +5,20 @@ import {
...
 import type {
   VirtualListItemProps,
   TreeItemProps,
   NamespaceItemProps,
 } from './tree-item';
-import { SmallIcon } from './icon-button';
-import type { Actions } from './constants';
+import { Actions } from './constants';
+
+const DBActionControls = ItemActionControls<Actions>;
```
Comment (lerouxb) [inline packages/compass-databases-navigation/src/collection-item.tsx:21]: @gribnoysup Is this the right way to make a specialised version of that function component to use in jsx below? Feels wrong to me, but can't really explain why.

Comment (gribnoysup) [reply packages/compass-databases-navigation/src/collection-item.tsx:21]: I think you might not even need to do this as TS should be able to infer the type from the actions that you are passing to the component. But also yes, support for this partial application was actually added not that long ago to typescript and that exactly is the use-case for something like this

Comment (lerouxb) [reply packages/compass-databases-navigation/src/collection-item.tsx:21]: oh I didn't even realise you can do this:

```
      <ItemActionControls<Actions>
        className={collectionActions}
        onAction={onAction}
        isActive={isActive}
        isHovered={isHovered}
        actions={actions}
        shouldCollapseActionsToMenu
      ></ItemActionControls>
```

Comment (lerouxb) [reply packages/compass-databases-navigation/src/collection-item.tsx:21]: ok it isn't necessary, then. Will remove.

File: packages/compass-sidebar/src/components/sidebar-title.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+} => ({
+  activeNamespace: state.databases.activeNamespace,
+  connectionInfo: state.connectionInfo.connectionInfo,
+});
+
+const connector = connect(mapStateToProps, {
+  globalAppRegistryEmit,
+});
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
```
Comment (gribnoysup) [inline packages/compass-sidebar/src/components/sidebar-title.tsx:92]: I know that it's a draft and you'd probably iterate more on this I would still want to highlight that I would recommend not defining component props based on connect mappers: when you do it that way you unintentionally create a very strong coupling between state and component. I would recommend going from the component implementation and looking for the minimal set of props required and then implementing connect to match it. In this case for example it seems like the only thing you want from connectionInfo is title, and so the component interface can be simplified and the `getConnectionTitle` part can be moved to the state mapper

Comment (lerouxb) [reply packages/compass-sidebar/src/components/sidebar-title.tsx:92]: Thinking about this more: Should I connect at all? might as well just pass in title, favorite and onAction from the sidebar?

Comment (gribnoysup) [reply packages/compass-sidebar/src/components/sidebar-title.tsx:92]: Generally speaking redux docs [strongly suggest](https://redux.js.org/style-guide/#connect-more-components-to-read-data-from-the-store) to connect as much components as possible and provide the data as granular as possible. I think this advice works great in most cases, even when it doesn't help much with rendering performance, but absolutely best when your whole app is following this. I'd say in this case it depends whether or not you are already using connection info and other connected props / methods somewhere a few levels higher in the react tree. If you do, then yeah, I think that would be fine to not connect

File: packages/compass-sidebar/src/components/sidebar-title.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  | 'open-instance-workspace'
+  | 'copy-connection-string'
+  | 'edit-favorite'
+  | 'cluster-info';
+
+const titleLabel = css({
+  overflow: 'hidden',
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  fontSize: '18px', // TODO: what's the right way to do this?
```
Comment (gribnoysup) [inline packages/compass-sidebar/src/components/sidebar-title.tsx:26]: You probably can use one of the heading elements from leafygreen

**Review by gribnoysup — APPROVED** [review]

File: packages/compass-sidebar/src/components/sidebar-title.tsx (hunk 4 above)
Comment (lerouxb) [reply packages/compass-sidebar/src/components/sidebar-title.tsx:26]: OK I'll try that in a separate PR - already building on this :)

**Pull request merged**